envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/server/CachedProtoResourcesSerializer.kt

```diff
@@ -30,14 +37,8 @@ internal class CachedProtoResourcesSerializer(
             .build<Collection<Message>, MutableCollection<Any>>()
     }
 
-    override fun serialize(resources: MutableCollection<out Message>): MutableCollection<Any> {
-        return if (reportMetrics) {
-            meterRegistry.timer("protobuf-cache.serialize.time")
-                .record(Supplier { getResources(resources) })
-        } else {
-            getResources(resources)
-        }
-    }
+    override fun serialize(resources: MutableCollection<out Message>): MutableCollection<Any> = serializeTimer
+        .record(Supplier { getResources(resources) })
```

Comment:
I think in recent version of Kotlin they fixed SAM so we won't need to wrap lambdas to get proper typing.

Reply:
yeah, we need to bump kotlin version.